boto3/s3/inject.py

```diff
@@ -356,3 +367,267 @@ def object_copy(self, CopySource, ExtraArgs=None, Callback=None,
         CopySource=CopySource, Bucket=self.bucket_name, Key=self.key,
         ExtraArgs=ExtraArgs, Callback=Callback, SourceClient=SourceClient,
         Config=Config)
+
+
+def upload_fileobj(self, Fileobj, Bucket, Key, ExtraArgs=None,
+                   Callback=None, Config=None):
+    """Upload a file-like object to S3.
+
+    This is a managed transfer which will perform a multipart upload in
+    multiple threads if necessary.
+
+    Usage::
+
+        import io
+        import boto3
+        data = io.BytesIO(b'foo')
+        s3 = boto3.client('s3')
+        s3.upload_fileobj(data, 'mybucket', 'mykey')
+
+    :type Fileobj: a file-like object
+    :param Fileobj: A file-like object to upload. At a minimum, it must
+        implement the `read` method.
+
+    :type Bucket: str
+    :param Bucket: The name of the bucket to upload to.
+
+    :type Key: str
+    :param Key: The name of the key to upload to.
+
+    :type ExtraArgs: dict
+    :param ExtraArgs: Extra arguments that may be passed to the
+        client operation.
+
+    :type Callback: method
+    :param Callback: A method which takes a number of bytes transferred to
+        be periodically called during the upload.
+
+    :type Config: boto3.s3.transfer.TransferConfig
+    :param Config: The transfer configuration to be used when performing the
+        upload.
+    """
+    subscribers = None
+    if Callback is not None:
+        subscribers = [ProgressCallbackInvoker(Callback)]
+
+    config = Config
+    if config is None:
+        config = TransferConfig()
+
+    with TransferManager(self, config) as manager:
+        future = manager.upload(
+            fileobj=Fileobj, bucket=Bucket, key=Key,
+            extra_args=ExtraArgs, subscribers=subscribers)
+        return future.result()
+
+
+def bucket_upload_fileobj(self, Fileobj, Key, ExtraArgs=None,
+                          Callback=None, Config=None):
+    """Upload a file-like object to this bucket.
+
+    This is a managed transfer which will perform a multipart upload in
+    multiple threads if necessary.
+
+    Usage::
+
+        import io
```

Comment:
Same thing here. I think it would be useful to add an open file example with 'wb'